File: old/alarm.java
import java.util.ArrayList;
import java.util.List;
import java.io.*;
import java.util.Collections;
import java.util.Date;
import java.util.Calendar;
import java.text.DateFormat;	
import java.text.SimpleDateFormat;


class alarm {
    

	class timeTable{
		List<List<taskObj>> dayTimeTable = new ArrayList<List<taskObj>>(24);
	
		//void readTimeTable


	};


	
	class taskObj	{
		String taskDescription;
		//Start time
		int hour;
		int minute;
		int duration;

		int endingHour;
		int endingMin;
		
		boolean isRecurring;
	};


	void printTimetable(int daynumber, List<taskObj> timetable){

		for(taskObj task : timetable){
			System.out.println(task.taskDescription);
		}


    taskObj getTime(String timeString){
		
		taskObj taskStartTime = new taskObj();
		String timeStr = new String(timeString);

		taskStartTime.hour = Integer.parseInt(timeStr.substring(0,timeStr.indexOf(':')));
		taskStartTime.minute = Integer.parseInt(timeStr.substring(timeStr.indexOf(':')+1));

		return taskStartTime;
	}

	String getDay(int daynumber){
	
		if(daynumber==1){return "Sunday";}
		if(daynumber==2){return "Monday";}
		if(daynumber==3){return "Tuesday";}
		if(daynumber==4){return "Wednesday";}
		if(daynumber==5){return "Thursday";}
		if(daynumber==6){return "Friday";}
		if(daynumber==7){return "Saturday";}
		else{return "ERROR";}
	}

/*
	void addTask(List<taskObj> timetable,String taskDescr, 
		String timeString, int taskDuration){

		int x;

		taskObj newTask = getTime(timeString);
		newTask.taskDescription = taskDescr;
		newTask.duration = taskDuration;
		
		timetable.add(newTask);
	}
*/





    public static void main(String[] args) {


    	alarm anAlarm = new alarm();
    	List<List<taskObj>> dayTimeTable = new ArrayList<List<taskObj>>(24);




/*			/////// USEFUL

    	taskObj asd = anAlarm.getTime("23:15");
    	System.out.println(asd.hour + " " + asd.minute);

        System.out.println("Hello World! " + anAlarm.test); // Display the string.



		DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
		Date date = new Date();
		System.out.println(dateFormat.format(date)); //2014/08/06 15:59:48

		Calendar c = Calendar.getInstance();
		c.setTime(date);
		int dayOfWeek = c.get(Calendar.DAY_OF_WEEK);
		System.out.println(anAlarm.getDay(dayOfWeek));

*/


    }
}
}

File: old/c cold/alarm.c
/* localtime example */
#include <stdio.h>
#include <time.h>
#include <string.h>
#include <stdlib.h>

char dayTimeTable[24][60];   //24 hours a day, 60 minutes 

struct timeObj
{
	int hour;
	int minute;
};

struct timeObj getTime(char timeString[]){
	
	struct timeObj taskStartTime;

	char temp[2];

	int x;
	for(x = 0; timeString[x] != ':'; x++){
		temp[x] = timeString[x];
	}
	taskStartTime.hour = atoi(temp);

	for( ;timeString[x] != ':'; x++){
		temp[x] = timeString[x];
	}
	taskStartTime.minute = atoi(temp);

	return taskStartTime;

}



void addTask(char task[], char timeString[], int length){
	int x;

	struct timeObj aTime = getTime(timeString);

	for(x=0;x<length;x++){
		dayTimeTable[aTime.hour][aTime.minute+x] = task;
	}

}


void getDay(int daynumber){
	
	if(daynumber==0){printf("Sunday");}
	if(daynumber==1){printf("Monday");}
	if(daynumber==2){printf("Tuesday");}
	if(daynumber==3){printf("Wednesday");}
	if(daynumber==4){printf("Thursday");}
	if(daynumber==5){printf("Friday");}
	if(daynumber==6){printf("Saturday");}
}

main()
{
	time_t rawtime;
	struct tm * timeinfo;

	time ( &rawtime );
	timeinfo = localtime ( &rawtime );

	getDay(timeinfo->tm_wday);
	printf ( ", %dth \n", timeinfo->tm_mday );
	printf ( "Time: %d:%d \n", timeinfo->tm_hour, timeinfo->tm_min );


	return 0;

}